apps/bandwagon/views.py

```diff
@@ -370,12 +370,15 @@ def ajax_list(request):
         return http.HttpResponseBadRequest()
 
     # Get collections associated with this user
-    collections = Collection.objects.publishable_by(request.amo_user)
-
-    for collection in collections:
-        # See if the collections contains the addon
-        if addon_id in collection.addons.values_list('id', flat=True):
-            collection.has_addon = True
+    has_addon = """
+        select 1 from addons_collections as ac
+            where ac.addon_id = %s and ac.collection_id = collections.id
+            limit 1"""
+
+    collections = Collection.objects.publishable_by(request.amo_user) \
+        .extra(
+            select={'has_addon': has_addon},
+            select_params=(addon_id,))
```

Comment:
We usually don't use `\`, you could (if you wanted to, I'm not saying you have to) write it:
```python
    collections = (Collection.objects.publishable_by(request.amo_user)
        .extra(select={'has_addon': has_addon}, select_params=(addon_id,)))
```
or
```python
   collections = (
        Collection.objects.publishable_by(request.amo_user)
        .extra(select={'has_addon': has_addon}, select_params=(addon_id,)))
```

Also I usually prefer using lists instead of tuples for one element, I find it easier to see at first glance that it's a sequence, not a single element:
```python
   collections = (Collection.objects.publishable_by(request.amo_user)
        .extra(select={'has_addon': has_addon}, select_params=[addon_id]))
```

This is all really nitpicking ;)